File: src/test/cpp/model/squaretest.cpp
// Copyright 2013-2021 Phokham Nonava
//
// Use of this source code is governed by the MIT license that can be
// found in the LICENSE file.

#include "model/square.h"
#include "model/file.h"
#include "model/rank.h"

#include "gtest/gtest.h"

using namespace pulse;

TEST(squaretest, testValues) {
	for (auto rank: rank::values) {
		for (auto file: file::values) {
			int square = square::valueOf(file, rank);

			EXPECT_EQ(file, square::getFile(square));
			EXPECT_EQ(rank, square::getRank(square));
		}
	}
}

File: pulse-java/src/test/java/com/fluxchess/pulse/java/model/SquareTest.java
/*
 * Copyright 2013-2023 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse.java.model;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class SquareTest {

	@Test
	void testValues() {
		for (int rank : Rank.values) {
			for (int file : File.values) {
				int square = Square.valueOf(file, rank);

				assertThat(Square.getFile(square)).isEqualTo(file);
				assertThat(Square.getRank(square)).isEqualTo(rank);
			}
		}
	}
}
